arch/x86/core/ia32/userspace.S

```diff
@@ -183,6 +183,7 @@ SECTION_FUNC(TEXT, z_x86_syscall_entry_stub)
 #endif /* CONFIG_X86_KPTI */
 
 	sti			/* re-enable interrupts */
+	nop
```

Comment:
Pretty sure this is safe?  I don't know all the situations that might touch the direction flag, and frankly am a little confused as to why this is here (I guess because syscalls might change it?  But then why clear vs. save/restore?).  Almost certainly not a problem, but this looks voodish so it should probably stay until someone who knows the code better can decide.